qiskit/quantum_info/operators/predicates.py

```diff
@@ -21,6 +21,27 @@
 RTOL_DEFAULT = 1e-5
 
 
+def _get_phase_difference(mat1, mat2, atol=ATOL_DEFAULT):
```

Comment:
How about instead adding an optional argument to `matrix_equal` that allows to return the phase difference? Since it's already computed there, no?

Reply:
I actually wanted to do this as well, but I don't know of a particularly clean and backward compatible way to do this. The best I can think of is something like this:
```python
def matrix_equal(..., props=None)
  ...
  if props is not None:
    props['global_phase'] = ...
```
Is this what you had in mind?

Reply:
On a second though, this approach is cleaner than my previous implementation, so done in [7cfc20c](https://github.com/Qiskit/qiskit/pull/11248/commits/7cfc20c2c79e2a7f02e1a4cc40eb63159f8b2934).